dashboard/modules/snapshot/snapshot_schema.json

```diff
@@ -0,0 +1,145 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://github.com/ray-project/ray/dashboard/modules/snapshot/snapshot_schema.json",
+  "type": "object",
+  "properties": {
```

Comment:
Best practice to include a description for each property.

common questions a user or developer may have:
1. for timestamps, what is the unit? s, ms, ns?
2. for objects, are there any assumptions i can make? ex/ string map? or any map?
3. what is this field used for? how is it different than other, similarly named fields? (`envVars` and `runtimeEnv`)